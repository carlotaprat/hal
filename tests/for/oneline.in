for key, value in map: something = key + value

pass