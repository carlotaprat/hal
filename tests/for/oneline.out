digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="FOR_STMT"];
  n1 [label="FOR_STMT"];
  n2 [label="PARAMS"];
  n2 [label="PARAMS"];
  n3 [label="key"];
  n4 [label="value"];
  n5 [label="FUNCALL"];
  n5 [label="FUNCALL"];
  n6 [label="map"];
  n7 [label="ARGS"];
  n8 [label="BLOCK"];
  n8 [label="BLOCK"];
  n9 [label="FUNCALL"];
  n9 [label="FUNCALL"];
  n10 [label="something"];
  n11 [label="ARGS"];
  n12 [label="="];
  n13 [label="+"];
  n13 [label="+"];
  n14 [label="FUNCALL"];
  n14 [label="FUNCALL"];
  n15 [label="key"];
  n16 [label="ARGS"];
  n17 [label="FUNCALL"];
  n17 [label="FUNCALL"];
  n18 [label="value"];
  n19 [label="ARGS"];
  n20 [label="FUNCALL"];
  n20 [label="FUNCALL"];
  n21 [label="pass"];
  n22 [label="ARGS"];

  n0 -> n1 // "BLOCK" -> "FOR_STMT"
  n1 -> n2 // "FOR_STMT" -> "PARAMS"
  n2 -> n3 // "PARAMS" -> "key"
  n2 -> n4 // "PARAMS" -> "value"
  n1 -> n5 // "FOR_STMT" -> "FUNCALL"
  n5 -> n6 // "FUNCALL" -> "map"
  n5 -> n7 // "FUNCALL" -> "ARGS"
  n1 -> n8 // "FOR_STMT" -> "BLOCK"
  n8 -> n9 // "BLOCK" -> "FUNCALL"
  n9 -> n10 // "FUNCALL" -> "something"
  n9 -> n11 // "FUNCALL" -> "ARGS"
  n8 -> n12 // "BLOCK" -> "="
  n8 -> n13 // "BLOCK" -> "+"
  n13 -> n14 // "+" -> "FUNCALL"
  n14 -> n15 // "FUNCALL" -> "key"
  n14 -> n16 // "FUNCALL" -> "ARGS"
  n13 -> n17 // "+" -> "FUNCALL"
  n17 -> n18 // "FUNCALL" -> "value"
  n17 -> n19 // "FUNCALL" -> "ARGS"
  n0 -> n20 // "BLOCK" -> "FUNCALL"
  n20 -> n21 // "FUNCALL" -> "pass"
  n20 -> n22 // "FUNCALL" -> "ARGS"

}

